File: src/main/java/page/JL.java
package page;

import common.util.lang.LocaleCenter.Binder;

import javax.swing.*;

public class JL extends JLabel implements LocComp {

	private static final long serialVersionUID = 1L;

	private final LocSubComp lsc;

	public JL() {
		lsc = new LocSubComp(this);
		setBorder(BorderFactory.createEtchedBorder());
	}

	public JL(Binder binder) {
		this();
		lsc.init(binder);
	}

	public JL(int i, String str) {
		this();
		lsc.init(i, str);
	}

	public JL(String str) {
		this(-1, str);
	}

	@Override
	public LocSubComp getLSC() {
		return lsc;
	}

}

File: src/page/JL.kt
package page

import common.util.lang.LocaleCenter
import javax.swing.BorderFactory
import javax.swing.JLabel

class JL() : JLabel(), LocComp {
    private val lsc: LocSubComp

    constructor(binder: LocaleCenter.Binder?) : this() {
        lsc.init(binder)
    }

    constructor(i: Int, str: String) : this() {
        lsc.init(i, str)
    }

    constructor(str: String) : this(-1, str)

    override fun getLSC(): LocSubComp {
        return lsc
    }

    companion object {
        private const val serialVersionUID = 1L
    }

    init {
        lsc = LocSubComp(this)
        border = BorderFactory.createEtchedBorder()
    }
}
